ape_safe/packages.py

```diff
@@ -0,0 +1,125 @@
+from enum import Enum
+from importlib import resources
+from typing import TYPE_CHECKING, Union
+
+import requests
+from ape.managers.project import ProjectManager
+from packaging.version import Version
+from pydantic import BaseModel
+
+if TYPE_CHECKING:
+    from ape.contracts import ContractContainer, ContractInstance
+
+with resources.as_file(resources.files(__package__).joinpath("manifests")) as manifest_folder:
+    SAFE_PACKAGE_BY_VERSION = {
+        Version(manifest_path.stem.lstrip("safe-v")): ProjectManager.from_manifest(manifest_path)
+        for manifest_path in manifest_folder.glob("safe-*.json")
+    }
+    MULTISEND_PACKAGE = ProjectManager.from_manifest(manifest_folder / "multisend.json")
+
+
+class PackageType(str, Enum):
+    SINGLETON = "SafeSingleton"
+    PROXY = "SafeProxy"
+    PROXY_FACTORY = "SafeProxyFactory"
+
+    def __call__(self, version: Union[Version, str]) -> "ContractContainer":
+        if not isinstance(version, Version):
+            version = Version(version.lstrip("v"))
+
+        if not (package := SAFE_PACKAGE_BY_VERSION.get(version)):
+            raise KeyError(f"Unknown version 'v{version}'.")
+
+        elif self is PackageType.PROXY_FACTORY:
+            if version == Version("1.1.1"):
+                return package.ProxyFactory
+
+            elif version <= Version("1.3.0"):
+                return package.GnosisSafeProxyFactory
+
+            else:
+                return package.SafeProxyFactory
+
+        elif version > Version("1.3.0"):
+            SafeSingleton = package.Safe
+
+        else:
+            SafeSingleton = package.GnosisSafe
+
+        if self is PackageType.SINGLETON:
+            return SafeSingleton
+
+        # NOTE: SafeProxy has `masterCopy() -> address` in it, vs. `SafeSingleton`
```

Comment:
I believe `masterCopy` is automatically added to the Safe contract type in Ape during proxy detection

Reply:
The idea was to bypass that detection, which I think also I should provide `proxy_info=` as well

Reply:
[96ace69](https://github.com/ApeWorX/ape-safe/pull/70/commits/96ace690d8bf0517d15fdee0e8ddf4380773aa9c)